### ISSUE TITLE
Trainable weight and scalar parameters in keras

### ISSUE THREAD

**Issue by ghost** [keras-team/keras#10988]
Consider the basic network of MNIST digit recognition.

My problem is to take input as the convolution feature maps from one layer and transform into smaller feature spaces f and g, where **f = Wf(x)** and **g = Wg(x)**, where Wf and Wg are **learned (trainable) weight matrices** of some particular dimensions ( dimensions are fixed and given ). Then the next problem after some steps is to multiply feature maps ( of another layer ) with a trainable scalar variable " **gamma** " where **gamma is initialized as 0**, The value of trainable parameters **Wf, Wg and gamma** - will keep on updating as per the final loss function of the network. The final loss function is a function of y_pred and y_true.

Is it possible to write in keras (with tensorflow backend) for this exact network ? 
Which layer / function should I use in keras for trainable Wf, Wg and gamma ? 
Which layer / function supports this trainable weight and scalar feature in tensorflow ?

Please help me on how to implement this task in keras and update these weight matrices and gamma ( the weight matrices of the other conv. layers will be also updated ) w.r.t the final loss function ? I am applying this network on RGB images, so the input images are of 3 channels.

Thanks for your time and help.

**Comment by yachee-gupta** [comment]
Hi @farizrahman4u, can you please help me in this problem and solve the issue. 
Thank you very much for your time and help

**Comment by yachee-gupta** [comment]
Hi @fchollet , please help me to write this part of the network in keras. Which methods/functions should I use in keras for the above part of the network. 
Thanks.
